client/webserver/webserver.go

```diff
@@ -0,0 +1,401 @@
+	mux.Get("/ws", s.handleWS)
+	// Webpages
+	mux.Get("/", s.handleHome)
+	mux.Get("/register", s.handleRegister)
+	mux.Get("/login", s.handleLogin)
+	mux.Get("/markets", s.handleMarkets)
+	mux.Get("/wallets", s.handleWallets)
+	mux.Get("/settings", s.handleSettings)
+	mux.Route("/api", func(r chi.Router) {
+		r.Use(middleware.AllowContentType("application/json"))
+		r.Post("/register", s.apiRegister)
+		r.Post("/login", s.apiLogin)
+	})
+	// Files
+	fileServer(mux, "/js", fp(root, "dist"))
+	fileServer(mux, "/css", fp(root, "dist"))
+	fileServer(mux, "/img", fp(root, "src/img"))
+	fileServer(mux, "/font", fp(root, "src/font"))
+
+	// Start serving.
+	listener, err := net.Listen("tcp", addr)
+	if err != nil {
+		log.Errorf("Can't listen on %s. web server quitting: %v", addr, err)
+		return
+	}
+	// Close the listener on context cancellation.
+	go func() {
+		<-ctx.Done()
+		err := listener.Close()
+		if err != nil {
+			log.Errorf("Problem shutting down rpc: %v", err)
+		}
+	}()
+	log.Infof("Web server listening on %s", listener.Addr())
+	err = httpServer.Serve(listener)
+	if err != http.ErrServerClosed {
+		log.Warnf("unexpected (http.Server).Serve error: %v", err)
+	}
+	log.Infof("Web server off")
+}
+
+// auth creates, stores, and returns a new auth token.
+func (s *webServer) auth() string {
+	// Create a token to identify the user. Only one token can be active at
+	// a time = only 1 authorized device at a time.
+	b := make([]byte, 32)
+	rand.Read(b)
+	token := hex.EncodeToString(b)
+	s.mtx.Lock()
+	s.authToken = token
+	s.mtx.Unlock()
+	return token
+}
+
+// token returns the current auth token.
+func (s *webServer) token() string {
+	s.mtx.RLock()
+	defer s.mtx.RUnlock()
+	return s.authToken
+}
+
+// watchMarket watches the specified market, cancelling any other market
+// subscriptions.
+func (s *webServer) watchMarket(dex, mkt string) (*core.OrderBook, error) {
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
+	old := s.openMarket
+	if old != nil {
+		log.Tracef("ending monitoring of market %s", old.market)
+		old.kill()
+	}
+	book, updates, err := s.core.Sync(dex, mkt)
+	if err != nil {
+		return nil, err
+	}
+	ctx, shutdown := context.WithCancel(s.ctx)
+	go func() {
+		log.Debugf("monitoring market %s @ %s", mkt, dex)
+	out:
+		for {
+			select {
+			case update := <-updates:
+				log.Tracef("order book update received for " + update.Market)
+			case <-ctx.Done():
+				break out
+			}
+		}
+		s.core.Unsync(dex, mkt)
+	}()
+	s.openMarket = &marketSyncer{
+		market: mkt,
+		kill:   shutdown,
+	}
+	return book, nil
+}
+
+// unmarket unsubscribes from any market being monitored.
+func (s *webServer) unmarket() {
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
+	if s.openMarket != nil {
+		log.Tracef("ending monitoring of market %s", s.openMarket.market)
+		s.openMarket.kill()
+	}
+	s.openMarket = nil
+}
+
+// decodeMarket decodes the market string into its two tickers.
+func decodeMarket(mkt string) (string, string, error) {
+	mkts := strings.Split(mkt, "-")
```

Comment:
We're using `"_"` in the server's market string representation: https://github.com/decred/dcrdex/blob/b6488b97693b1105f5e51a032d286a6437ecf005/dex/market.go#L20-L22
We can have a different representation client-side, of course, but I wanted to be sure this was intentional.

Reply:
Let's wait and see how this plays out. I think for displaying, I'd prefer the dash, and in this first implementation, the `mkt` string is pulled directly from the same source as the display value for simplicity. 

Eventually I'd expect the web app to have more information to send in this request, maybe even the asset IDs.

Reply:
Actually, I went ahead and changed over to asset IDs throughout.